### PR TITLE
Fix principal KeyError

### PR DESCRIPTION
**Pull request by bungoume**
* Fix KeyError: 'Principal' error
* and Reduce deep nest

```
2017-03-22 00:06:12,981 ERROR: Job "run_change_reporter (trigger: interval[0:15:00], next run at: 2017-03-22 00:12:55.849921)" raised an exception [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:520]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/apscheduler/scheduler.py", line 512, in _run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/usr/local/src/security_monkey/security_monkey/scheduler.py", line 35, in run_change_reporter
    reporter.run(account, interval)
  File "/usr/local/src/security_monkey/security_monkey/reporter.py", line 70, in run
    auditor.audit_these_objects(items_to_audit)
  File "/usr/local/src/security_monkey/security_monkey/auditor.py", line 135, in audit_these_objects
    method(item)
  File "/usr/local/src/security_monkey/security_monkey/auditors/s3.py", line 78, in check_policy
    self.inspect_policy_allow_all(statement, s3_item)
  File "/usr/local/src/security_monkey/security_monkey/auditors/s3.py", line 84, in inspect_policy_allow_all
    if statement['Principal'] == "*":
KeyError: 'Principal'
```

### CODE REVIEW

**Comment by monkeysecurity** [comment]
@mikegrima  - you were just playing with the S3 auditor.  Do you want to take a look at this one?

File: security_monkey/auditors/s3.py [hunk 1]
```diff
@@ -80,36 +80,28 @@ def check_policy(self, s3_item):
...
-                                for aws_entry in aws_entries:
-                                    if aws_entry[0:26] not in complained:
-                                        self.processCrossAccount(aws_entry, s3_item)
-                                        complained.append(aws_entry[0:26])
+            if statement.get('Effect') == 'Allow' and isinstance(statement.get("Principal"), dict):
+                aws_entries = statement["Principal"].get("AWS", [])
+                if isinstance(aws_entries, basestring):
+                    aws_entries = [aws_entries]
+                for aws_entry in aws_entries:
+                    if aws_entry[0:26] not in complained:
```
Comment (mikegrima) [inline security_monkey/auditors/s3.py]: Hello @bungoume : Can you elaborate what is happening here?

Comment (bungoume) [inline security_monkey/auditors/s3.py]: I'm sorry, I copied the previous implementation and I do not know the details.
I feel that it is better to remove it.

Comment (mikegrima) [inline security_monkey/auditors/s3.py]: Sounds good.

I did make some changes to the auditor that got merged in and is now causing conflicts.  Can you rebase your PR off of the latest develop and resolve the conflicts?

Comment (bungoume) [inline security_monkey/auditors/s3.py]: I resolved the conflict and rebased. Could you check it again, please?

Comment (mikegrima) [inline security_monkey/auditors/s3.py]: I'll try to look at this either tomorrow or Friday when I get some time. 

But it does look good -- want to do one last verification before hitting the merge button.

File: security_monkey/auditors/s3.py [hunk 2]
```diff
@@ -102,36 +102,29 @@ def check_policy(self, s3_item):
...
                 message = "POLICY - This Policy Allows Access From Anyone."
                 self.add_issue(10, message, s3_item)
                 return
 
-            if 'AWS' in statement['Principal']:
-                if statement['Principal']['AWS'] == "*":
-                    message = "POLICY - This Policy Allows Access From Anyone."
-                    self.add_issue(10, message, s3_item)
-                    return
+            if isinstance(principal, dict) and principal.get('AWS') == "*":
```
Comment (mikegrima) [inline security_monkey/auditors/s3.py]: To be clear -- this can be done in a follow-up PR.

**Pull request merged**